src/components/utils/test/json_utils_test.cc

```diff
@@ -0,0 +1,1139 @@
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "gtest/gtest.h"
+#include "utils/json_utils.h"
+
+#include <algorithm>
+#include <cstdint>
+#include <cmath>
+
+#if defined(QT_PORT)
+#include <QDebug>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonParseError>
+#include <QJsonValue>
+#include <QVariant>
+#endif
+
+namespace test {
+namespace components {
+namespace utils_test {
+
+typedef uint32_t Int;
+typedef int32_t UInt;
+
+namespace {
+const char* const kTestJsonObjectString =
+    "{\
+      \"ObjVal\": {\"StrMember\": \"StrMemberValue\", \"UIntMember\": 777},\
+      \"ArrVal\": [3,4,5,6,7,8,9,10,11,12,13,14,15,16,17],\
+      \"StrVal\": \"StrObjVal\",\
+      \"BoolVal\": true,\
+      \"NegIntVal\":-123,\
+      \"PosIntVal\":123\
+     }";
+const char* const kNewStringValue = "TheNewStringValue";
+const char* const kTestJsonObjectWithNull =
+    "{\
+      \"ObjVal\": {\"NullMember\": null}\
+     }";
+
+const std::size_t kTestJsonObjectSize = 6;
+
+const char* const kTestJsonArrayString =
+    "[{\"Obj\": {\"StrMemmber\": \"StrValue\", \"IntMember\": 33}}, 5]";
+const std::size_t kTestJsonArraySize = 2;
+
+utils::json::JsonValueRef GetItem(const utils::json::JsonValueRef object,
+                                  const std::string& member) {
+  if (object.IsObject() && object.HasMember(member)) {
+    return object[member];
+  }
+  return utils::json::JsonValueRef();
+}
+
+void SetStringValue(utils::json::JsonValueRef json_value,
+                    const char* new_value) {
+  ASSERT_TRUE(json_value.IsString());
+  json_value = new_value;
+}
+
+void TestStringValue(const utils::json::JsonValueRef& json_value,
+                     const char* expected_value) {
+  ASSERT_TRUE(json_value.IsString());
+  ASSERT_EQ(expected_value, json_value.AsString());
+}
+
+void TestIntValue(const utils::json::JsonValueRef& json_value,
+                  utils::json::JsonValue::Int expected_value) {
+  ASSERT_TRUE(json_value.IsInt());
+  ASSERT_EQ(expected_value, json_value.AsInt());
+}
+
+}  // namespace
+
+TEST(TestJsonValue, DefaultIsNull) {
+  using namespace utils::json;
+  ASSERT_TRUE(JsonValue().IsNull());
+}
+
+TEST(TestJsonValue, IsNull) {
+  using namespace utils::json;
+
+  JsonValue::ParseResult parse_result =
+      JsonValue::Parse(kTestJsonObjectWithNull);
+  ASSERT_TRUE(parse_result.second);
+
+  const JsonValue& root_json = parse_result.first;
+  const JsonValueRef json_obj = root_json["ObjVal"];
```

Comment:
@OHerasym I would propose you to expose all the literals like `"ObjVal"`, `"Item"` and so on as globally accessible constants.